### PR TITLE
use echarts tooltip

### LINKED ISSUE #37734: Summarization doesn't get removed when de-selecting it from chart settings
### Describe the bug

After summarizing with more than one metric and removing the metric from the chart setting, drill-down will still have the filter for one of the metrics and tooltip will show wrong information, while the graph doesn't show any reference to the metric

### To Reproduce

1. Create a GUI question from Orders, Summarize Count by Created At and Product category
2. 4 Lines show up in a line graph, that's cool
3. Now go to the chart settings and remove the Product category from the Series Breakout
4. Hover the line in the chart, see that a category is still applied (the values of the tooltip are not correct, it's summarized only by creation date), and if you drill-down, you'll see the category filter is applied to the graph

See that there is no reference to that category, but the graph still shows Doohickey in the tooltip, and when selected, "Doohickey" is applied as a filter
Screenshot 2024-01-16 at 9 44 09 AM

### Expected behavior

Maybe this is the expected behaviour, as there is a "Category" field in the summariation, but the grpah doesn't show any reference to that category, and the values are wrong

### Logs

_No response_

### Information about your Metabase installation

```json {
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.21+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.21",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.21+9",
    "os.name": "Linux",
    "os.version": "5.10.201-191.748.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "druid",
      "postgres",
      "redshift",
      "mysql",
      "bigquery-cloud-sdk",
      "mongo",
      "h2",
      "snowflake"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "14.7"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.1"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-01-16",
      "tag": "vUNKNOWN",
      "hash": "f926e90"
    },
    "settings": {
      "report-timezone": "US/Pacific"
    }
  }
}
```

### Severity

Can show misleading results

### Additional context

_No response_

### LINKED ISSUE #42114: [dc.js migration] Tooltip position sometimes is not correct
Details: [comment](https://github.com/metabase/metabase/issues/42114#issuecomment-2116433369)

---
https://metaboat.slack.com/archives/C01LQQ2UW03/p1714587141945559

### PR DESCRIPTION
**Pull request by alxnddr**
Closes https://github.com/metabase/metabase/issues/42114
Closes https://github.com/metabase/metabase/issues/37734

### Description

This PR replaces our old tooltip with the new ECharts tooltip. The scope of this PR is the [Milestone 1](https://www.notion.so/metabase/Migrate-tooltips-to-ECharts-for-more-predictable-behavior-364f2c71f3b248f1b839960af037279c?pvs=4#0a6e760a29d54395be150b8740859389).

**Changes to Line/Area/Bar/Combo tooltips**
- Previously users could see in tooltips all the columns for a given data point but now the tooltip shows only series data that is used on the chart.
- Non-stacked time series charts show the difference with the previous value for every series.
- Stacked charts tooltips now do not have limit of 9 series, the tooltip shows all of them. The sorting of series in tooltip by descending order was also removed.

**Changes to Waterfall and Scatter tooltips**
- Only new UI, nothing else should change.

**Changes to the Pie tooltip**
- Pie chart tooltips now do not have limit of 9 slices, the tooltip shows all of them. The sorting of slices in tooltip by descending order was also removed.

### How to verify

Verify the changes listed above.

### Demo

Pie chart
Screenshot 2024-08-14 at 8 39 14 PM

Stacked tooltip
Screenshot 2024-08-14 at 8 39 18 PM

Non-stacked tooltip
Screenshot 2024-08-14 at 8 39 56 PM

All columns tooltip
Screenshot 2024-08-14 at 8 40 10 PM

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by cdeweyx — APPROVED** [review]

**Review by kdoh — APPROVED** [review]

File: frontend/src/metabase/visualizations/echarts/cartesian/option/tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  chartModel: BaseCartesianChartModel,
+  settings: ComputedVisualizationSettings,
+): TooltipOption => {
+  return {
+    trigger: "item",
+    appendToBody: true,
+    confine: true,
+    className: TooltipStyles.ChartTooltipRoot,
+    formatter: params => {
+      const isAxisTooltip = Array.isArray(params);
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/cartesian/option/tooltip.tsx:56]: Sorry if I misunderstood the outcome of the [thread]([url](https://metaboat.slack.com/archives/C064QMXEV9N/p1723483422025159?thread_ts=1723223927.507779&cid=C064QMXEV9N)) and what the code is doing here, but did we decide to not use the axis tooltip for now? If so, do we still need this `isAxisTooltip` variable?

**Review by EmmadUsmani — COMMENTED** [review]
The code generally looks good but I ran into a few bugs:

- Pie charts are missing the total value



question is just count of products by category



happens in the other slice as well

- line chart sometimes included the total column, sometimes it didn't, and when it did it had a single `NaN%`





Question is a line chart on the below data set (no changes to the table in QB). X-axis is date, breakout by company, y-axis is profit.

[Nepal Hydropower Profit Loss.csv](https://github.com/user-attachments/files/16630879/Nepal.Hydropower.Profit.Loss.csv)

- Tooltip creates a vertical scroll bar in both QB and dashboards that lingers even after the tooltip is no longer shown

https://github.com/user-attachments/assets/eaeb9cc6-0266-4c59-aed1-b137a8be6e7e



You can see on this dashboard I have a double scroll bar now

- Not a bug, but now small dashcards are always covered by the tooltip. I think we need to reconsider the behavior here





Before the chart was at least partially visible, now it's completely blocked

**Comment by alxnddr** [comment]
@EmmadUsmani thanks for reviewing!

> Pie charts are missing the total value
Fixed this in the last commit

> line chart sometimes included the total column, sometimes it didn't, and when it did it had a single NaN%
Is it a time series? On time series the third column is the diff with the previous value, now we show it for all series. I'll check if `NaN%` is the correct value or a bug

**Comment by EmmadUsmani** [comment]
@alxnddr Yup it's timeseries

**Comment by alxnddr** [comment]
@EmmadUsmani I fixed the issues, thanks!

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/use-chart-events.ts [hunk 2]
```diff
@@ -69,6 +72,42 @@ export const useChartEvents = (
...
+
+      chartRef.current?.setOption({ yAxis: yAxisShowOption }, false, true);
+    },
+    [
+      chartModel.leftAxisModel,
+      chartModel.rightAxisModel,
+      chartRef,
+      hoveredSeriesDataKey,
+    ],
+  );
```
Comment (kulyk) [inline frontend/src/metabase/visualizations/visualizations/CartesianChart/use-chart-events.ts:109]: Doesn't seem like `updateYAxisVisibility` is actually used?

**Review by kulyk — APPROVED** [review]

**Review by EmmadUsmani — APPROVED** [review]
All bugs are fixed now, looks great! 🚢

**Pull request merged**